pkg/resources/block_storage/resource_volume.go

```diff
@@ -340,13 +349,16 @@ func (r *ResourceVolume) Read(ctx context.Context, req resource.ReadRequest, res
 
 	// Update state model.
 	state.Name = types.StringValue(volume.Name)
-	state.Size = types.Int64Value(volume.Size)
 	state.Blocksize = types.Int64Value(volume.Blocksize)
 	state.CreatedAt = types.StringValue(volume.CreatedAT.String())
 	state.State = types.StringValue(string(volume.State))
 
+	if !state.Size.IsNull() {
```

Comment:
Size is optional and all optional attributes have Null value if not defined in the config.
And if attribute is optional and not computed, we must not set value in state (otherwise terraform errors out).

Reply:
btw size is optional in API and when creating volume from snapshot API does not allow you to set size.